File: clojure/leap/leap.clj
(ns leap)

(defn divisible-by? [factor n]
  (zero? (mod n factor)))

(defn divisible-by-100-but-not-400 [year]
  (and (divisible-by? 100 year)
       (not (divisible-by? 400 year))))

(defn leap-year? [year]
  (and (divisible-by? 4 year)
       (not (divisible-by-100-but-not-400 year))))

File: go/leap/leap.go
package leap

func IsLeapYear(year int) bool {
	return IsDivisibleBy(4, year) &&
		!(IsDivisibleBy(100, year) && !IsDivisibleBy(400, year))
}

func IsDivisibleBy(divisor int, year int) bool {
	return year%divisor == 0
}
